evalml/pipelines/pipeline_base.py

```diff
@@ -709,6 +709,8 @@ def repr_component(parameters):
         parameters_repr = ", ".join(
             [
                 f"'{component}':{{{repr_component(parameters)}}}"
+                if component != "DFS Transformer"
+                else f"'{component}':{{}}"
```

Comment:
I'm worried about explicitly excluding information this way. Without making note of what features we're expecting, we're opening ourselves up to confusion/bugginess if a user provides the wrong features json in `generate_pipeline_code`. I'd be much more comfortable if we were able to add the string names of features, or something similar.

Reply:
I agree but I cannot do so for a couple reasons:

1. the `__repr__` method on a featuretools feature is not executable - it looks something like `<Feature 1>`. We need the parameters (in this case the feature definitions) to be executable for our pipeline `__repr__` and generating pipeline code so grabbing the feature name won't change the behavior.
2. there isn't an easy way to rebuild features from scratch. Talking to @rwedge there is a good amount of information the feature needs to know (including dataset info) that isn't easily serialized into a string. Maybe something for the future but I think this is a good intermediate measure for now.

So basically I cannot keep the behavior our pipelines `__repr__` method unless featuretools features do as well. Would you be more comfortable if I issue another warning here?

Reply:
Gotcha, I understand the challenge here. I'm still very worried here about the chance for something to go wrong, but throwing a warning will have to do, I guess.

I still wonder if there's a way to reduce risk. Can we offer a util to save and load featuretools features from/to the DFSTransformer directly, or save the feature names as a parameter for a sanity check? That sort of thing would probably be out of scope, but we could file a followup issue.

Reply:
>  Can we offer a util to save and load featuretools features from/to the DFSTransformer directly

thats what I'm doing in the pipeline example! Loading in the features and setting the parameters of the `DFSTransformer`.

I can give verifying the feature names a try.

Reply:
Ah, true! I suppose I was more focused on the "saving" part. New request: can we make sure we mention this behavior in the docs? We probably don't need a full example, just a mention of what's necessary and maybe the code for how to save the featuretools features.

Looking forward to seeing the results of trying to verify the feature names 😄